eng/pipelines/templates/steps/sync-directory.yml

```diff
@@ -6,30 +6,47 @@ parameters:
 
 steps:
+  - pwsh: |
+      New-Item -Path ${{ parameters.PRDataArtifactPath }} -ItemType File
+    displayName: Create PRData Artifact
+
+  - ${{ each repo in parameters.Repos }}:
+    - pwsh: |
+        git clone --branch ${{ parameters.BaseBranchName }} https://github.com/azure/${{ repo }}
+        $repoPath = "${{ repo }}/${{ parameters.DirectoryToSync }}"
+
+        if (!(Test-Path $repoPath)) { mkdir $repoPath }
+        Remove-Item -v -r $repoPath
+        Copy-Item -v -r $(Build.SourcesDirectory)/${{ parameters.DirectoryToSync }} $repoPath
+        Get-ChildItem -r $repoPath
+
+      displayName: Copy ${{ parameters.DirectoryToSync }} from azure-sdk-tools to ${{ repo }}
+      workingDirectory: $(System.DefaultWorkingDirectory)
+
+    - template: /eng/common/pipelines/templates/steps/create-pull-request.yml
+      parameters:
+        RepoName: ${{ repo }}
+        RepoOwner: Azure
+        PRBranchName: ${{ parameters.PRBranchName }}
+        BaseBranchName: ${{ parameters.BaseBranchName }}
+        CommitMsg: ${{ parameters.CommitMessage }}
+        PRTitle: ${{ parameters.PRTitle }}
+        PRBody: ${{ parameters.PRBody }}
+        PushArgs: -f
+        WorkingDirectory: $(System.DefaultWorkingDirectory)/${{ repo }}
+        ScriptDirectory: $(System.DefaultWorkingDirectory)/eng/common/scripts
+
+    - pwsh: |
+        $PRData = "Azure;${{ repo }};$(Submitted.PullRequest.Number)"
+        Add-Content -Path ${{ parameters.PRDataArtifactPath }} -Value $PRData
+      displayName: Write Sync PR Data to Artifact File
 
-- ${{ each repo in parameters.Repos }}:
   - pwsh: |
-      git clone --branch ${{ parameters.BaseBranchName }} https://github.com/azure/${{ repo }}
-      $repoPath = "${{ repo }}/${{ parameters.DirectoryToSync }}"
-
-      if (!(Test-Path $repoPath)) { mkdir $repoPath }
-      Remove-Item -v -r $repoPath
-      Copy-Item -v -r $(Build.SourcesDirectory)/${{ parameters.DirectoryToSync }} $repoPath
-      Get-ChildItem -r $repoPath
-
-    displayName: Copy ${{ parameters.DirectoryToSync }} from azure-sdk-tools to ${{ repo }}
-    workingDirectory: $(System.DefaultWorkingDirectory)
-
-  - template: /eng/common/pipelines/templates/steps/create-pull-request.yml
-    parameters:
-      RepoName: ${{ repo }}
-      PRBranchName: ${{ parameters.PRBranchName }}
-      BaseBranchName: ${{ parameters.BaseBranchName }}
-      CommitMsg: ${{ parameters.CommitMessage }}
-      PRTitle: ${{ parameters.PRTitle }}
-      PRBody: ${{ parameters.PRBody }}
-      PushArgs: -f
-      WorkingDirectory: $(System.DefaultWorkingDirectory)/${{ repo }}
-      ScriptDirectory: $(System.DefaultWorkingDirectory)/eng/common/scripts
+      $PRData = "Azure;azure-sdk-tools;$(System.PullRequest.PullRequestNumber)"
+      Add-Content -Path ${{ parameters.PRDataArtifactPath }} -Value $PRData
+    displayName: Write Tools PR Data to Artifact File
+    condition: and(succeeded(), eq(variables['Build.Reason'], 'PullRequest'))
```

Comment:
No need to condition this I would write this file all the time it still is useful for diagnosing issues.

Reply:
The problem is that the `$(System.PullRequest.PullRequestNumber)` variable will only be present if it is a pull request. So we don't need to write it (And we cannot) when the build is queued manually